src/test/rpc/DeliveredAmount_test.cpp

```diff
@@ -0,0 +1,342 @@
+    // ter success is true is the ter is tesSUCCESS partial is true if a
+    // partial payment;
+    void
+    adjCounters(bool success, bool partial)
+    {
+        if (!success)
+        {
+            ++numExpectedNotSet_;
+            return;
+        }
+        if (!afterSwitchTime_)
+        {
+            if (partial)
+                ++numExpectedAvailable_;
+            else
+                ++numExpectedSetUnavailable_;
+            return;
+        }
+        // normal case: after switch time & successful transaction
+        ++numExpectedAvailable_;
+    }
+
+public:
+    explicit CheckDeliveredAmount(bool afterSwitchTime)
+        : afterSwitchTime_(afterSwitchTime)
+    {
+    }
+
+    void
+    adjCountersSuccess()
+    {
+        adjCounters(true, false);
+    }
+
+    void
+    adjCountersFail()
+    {
+        adjCounters(false, false);
+    }
+    void
+    adjCountersPartialPayment()
+    {
+        adjCounters(true, true);
+    }
+
+    // After all the txns are checked, all the `numExpected` variables should be
+    // zero. The `checkTxn` function decrements these variables.
+    bool
+    checkExpectedCounters() const
+    {
+        return !numExpectedAvailable_ && !numExpectedNotSet_ &&
+            !numExpectedSetUnavailable_;
+    }
+
+    // Check if the transaction has `delivered_amount` in the metaData as
+    // expected from our rules. Decrements the appropriate `numExpected`
+    // variable. After all the txns are checked, all the `numExpected` variables
+    // should be zero.
+    bool
+    checkTxn(Json::Value const& t, Json::Value const& metaData)
+    {
+        if (t[jss::TransactionType].asString() != "Payment")
+            return true;
+
+        bool isSet = metaData.isMember(jss::delivered_amount);
+        bool isSetUnavailable = false;
+        bool isSetAvailable = false;
+        if (isSet)
+        {
+            if (metaData[jss::delivered_amount] != "unavailable")
+                isSetAvailable = true;
+            else
+                isSetUnavailable = true;
+        }
+        if (isSetAvailable)
+            --numExpectedAvailable_;
+        else if (isSetUnavailable)
+            --numExpectedSetUnavailable_;
+        else if (!isSet)
+            --numExpectedNotSet_;
+
+        if (isSet)
+        {
+            if (metaData.isMember("DeliveredAmount"))
+            {
+                if (metaData[jss::delivered_amount] !=
+                    metaData["DeliveredAmount"])
+                    return false;
+            }
+            else
+            {
+                if (afterSwitchTime_)
+                {
+                    if (metaData[jss::delivered_amount] != t[jss::Amount])
+                        return false;
+                }
+                else
+                {
+                    if (metaData[jss::delivered_amount] != "unavailable")
+                        return false;
+                }
+            }
+        }
+        else
+        {
+        }
```

Comment:
Empty block

Reply:
Fixed in `Remove empty block and reword comment` comment